### PR TITLE
Add jitting switch and bug fixes

### PR DESCRIPTION
**Pull request by talumbau**

### CODE REVIEW

**Comment by talumbau** [comment]
This pull request enables the capability to turn off jitting, so the code in `functions.py` will run as straight Python code. See discussion in #292. When I enabled this capability, it became clear that some errors that occur in the jitted code are silently passing through. So, this PR also fixes those errors. I have also created PR #618 so that anyone who desires to look at the errors in depth can do so, without the fixes I included as the second commit in this PR. The fixes to get everything working again are as follows:

 - I added additional values to a few parameter arrays in `current_law_policy.json`. This is because I discovered that when running certain tests, we have `MARS` values that go up to 6, and yet we are indexing parameter arrays that only go up to 4. In the current jitted code base in master, this means that we are referencing garbage data. 

- In `functions.py`, we are returning values that are never declared in the function definition. This is because the functions are jitted and numba is doing the following: it reads all of the variables in the Python code, then creates a small C-like function that declares these variables as the necessary type (with the default initialization of that variable type following the rules of the C language) and then this function is compiled. In pure Python, we end up running the functions through code paths where certain variables are not declared prior to the return statement. Therefore, I added variable declarations for certain functions with default values of 0, which is what those variables default to now anyway. This just makes explicit what is happening in the compiled code anyway.

This PR should not be merged as is, but rather I think someone with appropriate domain knowledge should look at my changes to `current_law_policy.json` and both input the proper additional values and the proper names for the columns (which I just call `mars_5` and `mars_6`). Or someone could just comment on this PR and I will update my code to reflect the correct names and values.

**Comment by martinholmer** [comment]
T.J. @talumbau,
Why are you undoing the work in pull request #606, which was merged into the master branch six days ago?  There was a reason for doing the rearrangement of all the EITC parameters, but I don't see a reason for undoing that rearrangement.  What's going on?

@MattHJensen

**Comment by talumbau** [comment]
Yep, that's an error, although I don't understand at this moment how I messed it up. I will fix it and either create a new PR or merge the proper changes in to this PR. If you or someone else could comment on the changes to `current_law_policy.json` though, that would be very much appreciated. Thanks!

**Comment by martinholmer** [comment]
T.J. @talumbau,
I'm confused by your assertion that the MARS values run from 1 to 6.  Here is the description of MARS from the 2009 IRS-SOI PUF documentation:
```
24. MARS Marital (Filing) Status:
Aggregated Return ..................................0
Single .............................................1
Married filing a joint return or Widow(er) with
   dependent child (surviving spouse)...............2
Married filing separately ..........................3
Head of household ..................................4

NOTE: For returns sampled above a 0.07 percent rate, 
returns filed with a marital status of head of household 
claiming no dependents have been converted to single.
```

Perhaps the Tax-Calculator code needs to fixed in a different way that in this pull request.

@MattHJensen @feenberg

**Comment by talumbau** [comment]
@martinholmer yes I thought it would be surprising to all. Therefore, I opened #618 so that folks could verify that this is true for themselves. Please sync with that branch, set `DO_JIT` to `False`, and attempt to run the tests. You will see the test failure.

**Comment by talumbau** [comment]
I conclude from your comment, then, that if I am seeing those values for `MARS` in our test suite, we are then testing with bad data. Is that correct?

**Comment by feenberg** [comment]
Years ago married separate was either 3 or 6 depending on whether the 
couple lived together.

dan

File: taxcalc/decorators.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 import ast
 import toolz
 
+DO_JIT=False
```
Comment (talumbau) [inline taxcalc/decorators.py]: Oops this should be `True`.

**Comment by talumbau** [comment]
closing in favor of #620 which has changes based off latest master.

**Pull request closed without merging**